map_model/src/pathfind/v2.rs

```diff
@@ -3,82 +3,181 @@
 //! things here will probably move into pathfind/mod.rs.
 
 use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use geom::Duration;
 
 use crate::pathfind::uber_turns::UberTurnV2;
-use crate::{DirectedRoadID, Map, Path, PathConstraints, PathRequest, PathStep, TurnID, UberTurn};
+use crate::{
+    DirectedRoadID, Map, MovementID, Path, PathConstraints, PathRequest, PathStep, TurnID, UberTurn,
+};
+
+/// One step along a path.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum PathStepV2 {
+    /// Original direction
+    Along(DirectedRoadID),
```

Comment:
Name suggestions? :D   Forward, Backward, Turn?

Reply:
I think `Along | Contraflow` is pretty good, but `Turn` might be a little problematic since we already have a similar but different concept of what a Turn is. Do you think `Movement(MovementId)` would be better?

Admittedly "Movement" isn't a great name, but if we don't like Movement as a name, I'd prefer to get rid of it consistently, rather than overload "Turn".